### PR TITLE
Fix and Test PclusterConfig behaviour at runtime

### PR DESCRIPTION
**Pull request by ddeidda**
- Fix AdditionalIamPoliciesParam to store loaded policies at runtime
- Fix get section by key and label when section label is updated
- Accept optional config_file parameter in pcluster_config_reader

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/tests/pcluster/config/test_section_cluster.py [hunk 1]
```diff
@@ -106,7 +107,7 @@ def test_cluster_section_from_250_cfn(mocker, cfn_params_dict, expected_section_
 @pytest.mark.parametrize(
     "cfn_params_dict, expected_section_dict",
     [
-        ({}, DefaultDict["cluster"].value),
+        ({}, utils.merge_dicts(DefaultDict["cluster"].value, {"additional_iam_policies": []})),
```
Comment (enrico-usai) [inline cli/tests/pcluster/config/test_section_cluster.py]: We could add another test for 250_cfn with batch as a scheduler to have a test with an additional_iam_policies with the AWSBatchPolicy inside.

File: cli/tests/pcluster/config/test_section_cluster.py [hunk 2]
```diff
@@ -438,7 +441,8 @@ def test_cluster_section_from_231_cfn(mocker, cfn_params_dict, expected_section_
                     "initial_queue_size": 0,
                     "max_queue_size": 3,
                     "placement": "cluster",
-                    "base_os": "ubuntu1404",  # NOTE: We create the config with the old base_os (no longer supported)
+                    "base_os": "ubuntu1404",
```
Comment (enrico-usai) [inline cli/tests/pcluster/config/test_section_cluster.py]: I'd keep the comment. It was to highlight we have an unsupported os in the cfn params

File: cli/tests/pcluster/config/utils.py [hunk 3]
```diff
@@ -72,6 +72,17 @@ def assert_param_validator(mocker, config_parser_dict, expected_error=None, caps
...
     mocker.patch.object(PclusterConfig, "_PclusterConfig__check_account_capacity")
 
+    # Mock IAM policies validator to prevent boto3 calls in tests
+    accepted_policies = [
+        "arn:aws:iam::aws:policy/CloudWatchAgentServerPolicy",
+        "arn:aws:iam::aws:policy/AWSBatchFullAccess",
+    ]
+
+    def mock_iam_policies_validate(self):
+        assert set(self.value).issubset(accepted_policies)
```
Comment (enrico-usai) [inline cli/tests/pcluster/config/utils.py]: could you add a comment to explain what are you doing with this function?

File: cli/tests/pcluster/config/test_section_cluster.py (hunk 1 above)
Comment (ddeidda) [reply cli/tests/pcluster/config/test_section_cluster.py]: From Cfn we load directly only existing policies

File: cli/pcluster/config/iam_policy_rules.py [hunk 4]
```diff
@@ -12,6 +12,7 @@
 import abc
 import sys
 
+import pcluster
```
[the remark this answers is not in the document]
Comment (ddeidda) [reply cli/pcluster/config/iam_policy_rules.py]: OK

[the remark this answers is not in the document]
Comment (ddeidda) [reply cli/pcluster/config/iam_policy_rules.py]: There's a circular dependency here; I moved the import inside the function. Please let me know if it is acceptable

File: cli/pcluster/config/iam_policy_rules.py [hunk 5]
```diff
@@ -51,17 +52,21 @@ class CloudWatchAgentServerPolicyInclusionRule(EC2IAMPolicyInclusionRule):
...
+    def policy_is_required(cls, config_parser, cluster_section_name):
         """Describe whether the policy represented by this class must be included."""
-        cw_log_settings = pcluster_config.get_section("cluster").get_param("cw_log_settings")
-        if cw_log_settings.value is not None:
+        cw_log_settings = None
+        should_include_policy = pcluster.config.mappings.CW_LOG.get("params").get("enable").get("default")
+
+        if config_parser.has_option(cluster_section_name, "cw_log_settings"):
+            cw_log_settings = config_parser.get(cluster_section_name, "cw_log_settings")
+        if cw_log_settings is not None:
```
[the remark this answers is not in the document]
Comment (ddeidda) [reply cli/pcluster/config/iam_policy_rules.py]: OK

**Pull request closed without merging**